File: app/src/main/java/tk/alexlopez/sallefy/activities/TracksListActivity.java
package tk.alexlopez.sallefy.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import tk.alexlopez.sallefy.R;
import tk.alexlopez.sallefy.adapters.TrackListAdapter;
import tk.alexlopez.sallefy.models.Playlist;
import tk.alexlopez.sallefy.models.Track;
import tk.alexlopez.sallefy.network.callback.TrackCallback;
import tk.alexlopez.sallefy.network.manager.TrackManager;

public class TracksListActivity extends AppCompatActivity implements TrackCallback {

    private RecyclerView mRecyclerView;
    private ArrayList<Track> mTracks;
    private TextView mPlaylistsTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tracks_list);

        iniViews();
        getData();
    }

    private void iniViews() {
        mRecyclerView = findViewById(R.id.recycler_view);
        mPlaylistsTitle = findViewById(R.id.playlist_title);
        LinearLayoutManager manager = new LinearLayoutManager(this, RecyclerView.VERTICAL, false);
        TrackListAdapter adapter = new TrackListAdapter(this, null);
        mRecyclerView.setLayoutManager(manager);
        mRecyclerView.setAdapter(adapter);
    }

    private void getData() {

        Intent intent = getIntent();
        int id = intent.getIntExtra("id", -1);

        if (id != -1) {
            TrackManager.getInstance(this).getAllTracksByPlaylistId(id, this);
        } else {
            //TrackManager.getInstance(this).getAllTracks(this);
            // TODO: Error
        }

        mTracks = new ArrayList<>();

    }

    @Override
    public void onTracksReceived(List<Track> tracks) {
        mTracks = (ArrayList) tracks;
        TrackListAdapter adapter = new TrackListAdapter(this, mTracks);
        mRecyclerView.setAdapter(adapter);
    }

    public void onTracksReceivedByPlaylistId(Playlist playlist) {
        mTracks = (ArrayList) playlist.getTracks();
        TrackListAdapter adapter = new TrackListAdapter(this, mTracks);
        mRecyclerView.setAdapter(adapter);
    }

    @Override
    public void onNoTracks(Throwable throwable) {

    }

    @Override
    public void onPersonalTracksReceived(List<Track> tracks) {

    }

    @Override
    public void onUserTracksReceived(List<Track> tracks) {

    }

    @Override
    public void onFailure(Throwable throwable) {

    }
}

File: app/src/main/java/tk/alexlopez/sallefy/activities/TracksListActivity.kt
package tk.alexlopez.sallefy.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_tracks_list.*
import tk.alexlopez.sallefy.R
import tk.alexlopez.sallefy.adapters.TrackListAdapter
import tk.alexlopez.sallefy.models.Playback
import tk.alexlopez.sallefy.models.Playlist
import tk.alexlopez.sallefy.models.Track
import tk.alexlopez.sallefy.models.User
import tk.alexlopez.sallefy.network.callback.TrackCallback
import tk.alexlopez.sallefy.network.manager.TrackManager
import java.util.*

class TracksListActivity : AppCompatActivity(), TrackCallback {

    private lateinit var mTracks: ArrayList<Track>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tracks_list)
        iniViews()
        data
    }

    private fun iniViews() {
        val manager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        val adapter = TrackListAdapter(this, null, playListId)
        recycler_view.layoutManager = manager
        recycler_view.adapter = adapter
    }

    private val playListId: Int
        get() = intent.getIntExtra("id", -1)



    private val data: Unit
        get() {
            val intent = intent
            val id = intent.getIntExtra("id", -1)
            if (id != -1) {
                TrackManager.getInstance(this).getAllTracksByPlaylistId(id, this)
            }
            mTracks = ArrayList()
        }

    override fun onTracksReceived(tracks: List<Track>) {
        mTracks = tracks as ArrayList<Track>
        val adapter = TrackListAdapter(this, mTracks, playListId)
        recycler_view?.adapter = adapter
    }

    override fun onTracksReceivedByPlaylistId(playlist: Playlist) {
        // set title
        playlist_title?.text = playlist.name
        mTracks = playlist.tracks as ArrayList<Track>
        val adapter = TrackListAdapter(this, mTracks, playListId)
        recycler_view?.adapter = adapter
    }

    override fun onResume() {
        super.onResume()
        data
    }

    override fun onNoTracks(throwable: Throwable) {}
    override fun onLikedTrack(response: Boolean?) {

    }

    override fun onUserInfoReceived(body: User?) {

    }

    override fun onPlaybackReceived(body: MutableList<Playback>?) {

    }

    override fun onPersonalTracksReceived(tracks: List<Track>) {}
    override fun onPlaylistsReceived(playlists: MutableList<Playlist>?) {

    }

    override fun onUserTracksReceived(tracks: List<Track>) {}
    override fun onCreateTrack() {}
    override fun onNoLikedTrack(response: Boolean?) {

    }

    override fun onTrackSelected(track: Track) {}
    override fun onTrackSelected(index: Int) {}
    override fun onPlaylistUpdated(response: Boolean?) {

    }

    override fun onFailure(throwable: Throwable) {}
}

